### PR TITLE
Miner blacklist

### PR DESCRIPTION
**Pull request by andreea-popescu-reef**
added admin panel  new job creation triggering job run
blacklist for miner hotkey

### CODE REVIEW

File: validator/app/src/compute_horde_validator/validator/management/commands/connect_facilitator.py [hunk 1]
```diff
@@ -322,186 +233,18 @@ async def handle_message(self, raw_msg: str | bytes):
...
-        if not neurons:
-            raise ValueError(f'Miner with {hotkey=} not present in this subnetowrk')
-        return neurons[0].axon_info
-
     async def miner_driver(self, job_request: JobRequest):
-        """ drive a miner client from job start to completion, the close miner connection """
-
+        """ drive a miner client from job start to completion, then close miner connection """
         miner, _ = await Miner.objects.aget_or_create(hotkey=job_request.miner_hotkey)
-        miner_axon_info = await self.get_miner_axon_info(job_request.miner_hotkey)
```
Comment (mzukowski-reef) [inline validator/app/src/compute_horde_validator/validator/management/commands/connect_facilitator.py:336]: you are breaking debug_connect_facilitator that allows testing facilitator, validator, miner, executor setup locally without any subtensor context

File: validator/app/src/compute_horde_validator/validator/admin.py [hunk 2]
```diff
@@ -18,12 +19,27 @@ def has_change_permission(self, *args, **kwargs):
...
+        if self.fields:
+            self.fields['miner'].queryset = Miner.objects.exclude(minerblacklist__isnull=False)
+
+class AddOnlyAdminJobRequestAdmin(AddOnlyAdmin):
+    form = AdminJobRequestForm
+    exclude = ['env'] # not used ?
+    list_display = ['uuid', 'docker_image', 'use_gpu', 'miner', 'created_at']
+    ordering = ['-created_at']
+
+class JobReadOnlyAdmin(AddOnlyAdmin):
```
Comment (mzukowski-reef) [inline validator/app/src/compute_horde_validator/validator/admin.py:38]: I strongly feel it is time for class ReadOnlyAdmin(AddOnlyAdmin) :D

File: validator/app/src/compute_horde_validator/validator/apps.py [hunk 3]
```diff
@@ -19,8 +20,15 @@ def maybe_create_default_admin(sender, **kwargs):
...
     name = 'compute_horde_validator.validator'
 
     def ready(self):
+        from django.db.models import signals
+
+        from .models import AdminJobRequest
+        from .tasks import trigger_run_admin_job_request
+
         post_migrate.connect(maybe_create_default_admin, sender=self)
+        signals.post_save.connect(receiver=trigger_run_admin_job_request, sender=AdminJobRequest)
```
Comment (mzukowski-reef) [inline validator/app/src/compute_horde_validator/validator/apps.py:34]: violation of  Restricted patterns / forbidden technologies: Django signals

https://github.com/reef-technologies/handbook/blob/master/docs/Code_conventions.md?plain=1#L106

File: validator/app/src/compute_horde_validator/validator/models.py [hunk 4]
```diff
@@ -49,6 +59,26 @@ class Meta:
...
+    env = models.JSONField(blank=True, default=dict, help_text="environment variables for the job")
+
+    use_gpu = models.BooleanField(default=False, help_text="Whether to use GPU for the job")
+    input_url = models.URLField(blank=True, help_text="URL to the input data source", max_length=1000)
+    output_url = models.TextField(blank=True, help_text="URL for uploading output")
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    def get_args(self):
+        return self.args.split()
```
Comment (mzukowski-reef) [inline validator/app/src/compute_horde_validator/validator/models.py:84]: shlex.split

**Review by mzukowski-reef — CHANGES_REQUESTED** [review]

File: validator/app/src/compute_horde_validator/validator/management/commands/connect_facilitator.py (hunk 1 above)
Comment (andreea-popescu-reef) [reply validator/app/src/compute_horde_validator/validator/management/commands/connect_facilitator.py:336]: fixed now

**Review by mzukowski-reef — APPROVED** [review]

**Pull request merged**